geo/src/algorithm/sweep/tests.rs

```diff
@@ -0,0 +1,583 @@
+use std::iter::FromIterator;
+
+/// Helper function to compute all intersections using brute force
+fn compute_brute_force_intersections<T: GeoFloat>(
+    lines: &[Line<T>],
+) -> Vec<(Line<T>, Line<T>, LineIntersection<T>)> {
+    let mut result = Vec::new();
+
+    for i in 0..lines.len() {
+        for j in (i + 1)..lines.len() {
+            // Also check if lines are exactly the same - skip duplicates
+            if lines[i] == lines[j] {
+                continue;
+            }
+
+            // Use the standard line_intersection function for all cases
+            let intersection = line_intersection(lines[i], lines[j]);
+
+            if let Some(intersection) = intersection {
+                // Include all types of intersections
+                result.push((lines[i], lines[j], intersection));
+            }
+        }
+    }
+    result
+}
+
+/// Helper function to verify that sweep line and brute force find the same intersections
+fn verify_intersections(lines: &[Line<f64>]) {
+    // Get intersections using both algorithms
+    let sweep_intersections: Vec<_> =
+        Intersections::<_>::from_iter(lines.iter().cloned()).collect();
+    let brute_force_intersections = compute_brute_force_intersections(lines);
+
+    // Check for same count
+    assert_eq!(
+        sweep_intersections.len(),
+        brute_force_intersections.len(),
+        "Sweep algorithm found {} intersections but brute force found {}",
+        sweep_intersections.len(),
+        brute_force_intersections.len()
+    );
+
+    // Check that all brute force intersections are found by sweep
+    // and that their intersection details match
+    for (bf_line1, bf_line2, bf_intersection) in &brute_force_intersections {
+        let matching_intersection = sweep_intersections.iter().find(|(line1, line2, _)| {
+            (*line1 == *bf_line1 && *line2 == *bf_line2)
+                || (*line1 == *bf_line2 && *line2 == *bf_line1)
+        });
+
+        assert!(
+            matching_intersection.is_some(),
+            "Sweep algorithm missed an intersection between lines"
+        );
+
+        // Compare the actual intersection details
+        #[allow(clippy::assertions_on_constants)]
+        if let Some((_, _, sw_intersection)) = matching_intersection {
+            match (bf_intersection, sw_intersection) {
+                (
+                    LineIntersection::SinglePoint {
+                        intersection: bf_point,
+                        is_proper: bf_proper,
+                    },
+                    LineIntersection::SinglePoint {
+                        intersection: sw_point,
+                        is_proper: sw_proper,
+                    },
+                ) => {
+                    assert_eq!(
+                        *bf_proper, *sw_proper,
+                        "Proper intersection status doesn't match for the same line pairs"
+                    );
+
+                    assert_relative_eq!(
```

Comment:
As a rule, I'm pretty happy to accept a little slop in our floating point math.

Just for my own understanding, in this case we use the same algorithm for computing the value of the intersection - so it should actually match exactly. Right? I'm fine to leave the slop in, just making sure I understand.